File: server/models/measure_type.js
const { Schema, model } = require("mongoose");

const measureTypeSchema = Schema({
  name: { type: String, required: true, trim: true },
  text: { type: String, required: true, trim: true },
  status: {
    type: String,
    required: true,
    enum: ["active", "inactive", "delete"],
  },
  dtStatus: { type: Date, default: Date.now() },
  dtCr: { type: Date, default: Date.now() },
  dtUp: Date,
});

measureTypeSchema.index({ name: 1 }, { unique: true });

measureTypeSchema.set("toObject", { virtuals: true });
measureTypeSchema.set("toJson", { virtuals: true });

exports.MeasureType = model("MeasureType", measureTypeSchema);

File: marina/lib/common/models/measure_type.dart
class MeasureType {
  String? id;
  String? name;
  String? text;

  MeasureType({this.id, this.name, this.text});

  factory MeasureType.fromJson(Map<String, dynamic> json) =>
      MeasureType(id: json["_id"], name: json["name"], text: json["text"]);

  Map<String, dynamic> toJson() => {"_id": id, "name": name, "text": text};
}
